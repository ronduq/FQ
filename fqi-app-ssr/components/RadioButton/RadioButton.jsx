import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import s from './RadioButton.module.scss';

class RadioButton extends React.Component {

  handleRadioButtonChange = (event) => {
    const {
      changeAction
    } = this.props;
    const criteria = event.target.value;
    if (criteria === 'quality'){
      changeAction(criteria, 'topToBottom');
    } else {
      changeAction(criteria, 'bottomToTop');
    }
  }

  render() {
    const {
      label,
      selected,
      id,
      radioGroup

    } = this.props;

    return (
      <div className={s.radio}>
        <input 
          type="radio" 
          id={id} 
          name={radioGroup}
          value={id} 
          onChange={this.handleRadioButtonChange}
          checked={selected}
          tabIndex='0'
          autoComplete="off"
        />
        <label htmlFor={id}>{label}</label>
      </div>
    );
  }
};

RadioButton.propTypes = {
  label: PropTypes.string,
  checked: PropTypes.bool,
  type: PropTypes.string,
  radioGroup: PropTypes.string,
};

export default RadioButton;