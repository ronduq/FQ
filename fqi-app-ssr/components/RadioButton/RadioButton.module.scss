
@import '../../style/settings/colours';
@import '../../style/settings/fonts';
@import '../../style/settings/breakpoints';

.radio {
  position: relative;
  margin-bottom: 0.5em;
  line-height: 20px;
  display: block;
  margin-left: 10px;
  margin-right: 5px;
  cursor: pointer;
  @include respond-to('tablet', min-width) {
    display: inline-block;
  }
}
fieldset :last-child {
  margin-bottom: 0;
}
input[type="radio"] {
  position: absolute;
  width: 20px;
  height: 20px;
  overflow: hidden;
  margin: 0;
  padding: 0;
  border: 0;
  outline: 0;
  opacity: 0;
}
input[type="radio"] + label:before {
  display: inline-block;
  width: 20px;
  height: 20px;
  margin-right: 4px;
  content: " ";
  vertical-align: top;
  background: $colour-grey-light;
  border: 1px solid $colour-grey-medium;
  border-radius: 50%;
  transform: translateY(-2px);
  cursor: pointer;
}
input[type="radio"] + label {
  @include font-size('medium');
  line-height: 20px;
}
input[type="radio"]:hover + label:before {
  border: 1px solid $colour-secondary;
}
input[type="radio"]:checked:focus + label:before{
  box-shadow: 0 0px 0px 4px white inset, 0 0 0 2px dodgerblue;
}
input[type="radio"]:checked + label:before {
  background: $colour-secondary;
  box-shadow: 0 0px 0px 4px white inset;
  border: 1px solid $colour-secondary;
}
input[type="radio"]:checked + label{
  @include font-family('proxima-bold');
}
