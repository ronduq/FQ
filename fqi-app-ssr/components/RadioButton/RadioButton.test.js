/* eslint-env jest */

import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import RadioButton from './RadioButton'


describe('With Snapshot Testing', () => {
  const changeAction = jest.fn();
  const updateItem = jest.fn();
  const selectAllItemsRadioButton = jest.fn();
  const selectAllItems = jest.fn();

  const props = {
    label: 'Quality',
    id: 'quality',
    changeAction: changeAction,
    selected: true
  }
  

  it('RadioButton snapshot"', () => {
    const component = renderer.create(<RadioButton />)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  })

  it('should fire handleRadioButtonChange', () => {
    const wrapper = mount(<RadioButton {...props} />);
    const event = {target: {checked: true}}
    wrapper.find('input').simulate('change', event);
    expect(changeAction).toBeCalledTimes(1);
  })

  it('should fire handleRadioButtonChange for quality', () => {
    const wrapper = mount(<RadioButton {...props} />);
    const event = {target: {value: 'quality'}}
    wrapper.find('input').simulate('change', event);
    expect(changeAction).toBeCalledTimes(2);
  })
});
