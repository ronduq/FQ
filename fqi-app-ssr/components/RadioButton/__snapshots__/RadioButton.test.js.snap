// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing RadioButton snapshot" 1`] = `
<div
  className="radio"
>
  <input
    autoComplete="off"
    onChange={[Function]}
    tabIndex="0"
    type="radio"
  />
  <label />
</div>
`;
